### PR TITLE
Minimize use of global vars on gulp task callbacks

### PR DESCRIPTION
**Pull request by davrodpin**

### CODE REVIEW

File: bin/grommet.js [hunk 1]
```diff
@@ -17,76 +18,120 @@ String.prototype.capitalize = function() {
   return words.join(' ');
 };
 
-function nodeVersionSupported() {
-  return Number(process.version.match(/^v(\d+\.\d+)/)[1]) >= 0.10;
+function fileExists(filePath) {
+  try {
+      result = fs.statSync(filePath);
```
Comment (alansouzati) [inline bin/grommet.js]: can we keep the two-character spacing in here? I believe we have 4, where it is supposed to be two.

I need to add the jslint in this file as well. so my apologies.

**Comment by davrodpin** [comment]
@alansouzati, done.

File: bin/grommet.js [hunk 2]
```diff
@@ -17,76 +18,120 @@ String.prototype.capitalize = function() {
...
-      console.error(
-        '[grommet] Currently you have Node ' + process.version +
-        ' and NPM ' + npmVersion
-      );
-      process.exit(1);
-    }
+function dependenciesSupported(nodeVersion, npmVersion) {
+  if (!nodeVersionSupported(nodeVersion) || !npmVersionSupported(npmVersion)) {
+    console.error(
+      '[grommet] Grommet requires Node v'+supportedNodeVersion+
```
Comment (alansouzati) [inline bin/grommet.js]: can we add space in here? as you have for the other console messages?

'[grommet] Grommet requires Node v' + supportedNodeVersion +

File: bin/grommet.js [hunk 3]
```diff
@@ -17,76 +18,120 @@ String.prototype.capitalize = function() {
...
+      ' and NPM ' + npmVersion
+    );
+    return false;
+  }
+
+  return true;
+}
+
+function copyAssetsAndInstall(from, to, command) {
+  var installDependencies = function() {
```
Comment (alansouzati) [inline bin/grommet.js]: i like that!

File: bin/grommet.js [hunk 4]
```diff
@@ -17,76 +18,120 @@ String.prototype.capitalize = function() {
...
+          gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        }
+      })
+    }
   }
 };
 
+var supportedNodeVersion = '0.10';
+var supportedNpmVersion = '1.4';
```
Comment (alansouzati) [inline bin/grommet.js]: We recently updated our support stack to 3+ for NPM. would you be kind enough to update here ?

**Comment by alansouzati** [comment]
I just had a chance to test this PR, and it looks great. it has only one minor fix needed before I go ahead and approve it. thanks for your contribution.

**Comment by davrodpin** [comment]
> I just had a chance to test this PR, and it looks great. it has only one minor fix needed before I go ahead and approve it. thanks for your contribution.

Hi @alansouzati, I believe I have fixed the minor issues you have pointed on your previous comments. Could you please take a look again? Thank you!

**Comment by alansouzati** [comment]
thank you so much :)

**Pull request merged**